src/utarray.h

```diff
@@ -38,8 +38,8 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define UTARRAY_UNUSED
 #endif
 
-#ifndef oom
-#define oom() exit(-1)
+#ifndef utarray_oom
+#define utarray_oom() exit(-1)
 #endif
```

Comment:
Is anyone out there using `-Doom()=whatever` today? If so, we might need to add something like

    #ifdef oom
    #error "The name of macro 'oom' has been changed to 'utarray_oom'. Please update your code."
    #endif

Does anyone reading this PR want to speak up and say "I've used `-Doom()=`?"
I'm inclined to leave this PR open for comments for a month, and then if nobody complains, merge it without any `#ifdef oom` safety net. `oom` was a terrible name for a macro.

Reply:
How about add them for now and remove them after a month if no one complains?

Reply:
Well, if I add the `#ifdef oom` safety net, I'm going to leave it in at least for one release.
I have no real sense of how many people use uthash straight from github master, versus from distributions that update `/usr/include/uthash/uthash.h`, versus vendoring it into their codebase and updating it once every couple of years, if ever.

But I guess adding the safety net is the right thing to do.

Reply:
OK! Please see my updated commit.